python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -331,3 +332,71 @@ def get_units(value):
         return _unit_map[value.lower()]
 
     return value
+
+
+class Container(object):
+    def __init__(self, arg):
+        self.arg = arg
+
+
+class DateOffset(pd.DateOffset, Container):
+    def __init__(self, n=1, normalize=False, **kwds):
+        if normalize:
+            raise NotImplementedError(
+                "normalize not yet supported for DateOffset"
+            )
+
+        # TODO: Pandas supports combinations
+        if len(kwds) > 1:
+            raise NotImplementedError(
+                "only a single unit may" "be specified at a time"
+            )
+
+        all_possible_kwargs = {
+            "years",
+            "months",
+            "weeks",
+            "days",
+            "hours",
+            "minutes",
+            "seconds",
+            "microseconds",
+            "nanoseconds",
+            "year",
+            "month",
+            "week",
+            "day",
+            "hour",
+            "minute",
+            "second",
+            "microsecond",
+            "nanosecond",
+        }
+
+        supported_kwargs = {"months"}
+
+        for k, v in kwds.items():
+            if k in all_possible_kwargs:
+                # Months must be int16
+                dtype = "int16" if k == "months" else None
+                kwds[k] = cudf.Scalar(v, dtype=dtype)
```

Comment:
What does using `cudf.Scalar` objects do to the `__repr__`?

Reply:
```>>> cudf.DateOffset(months=1)
<DateOffset: months=Scalar(1, dtype=int16)>
```